lib/client.rb

```diff
@@ -275,10 +281,18 @@ def http_transport(message)
       'Content-Type' => 'application/json',
       'M-Api-Token' => @config[:token]
     }
-    response = @http.send_request('PUT', '/tel/v2.0/metrics', message, headers)
 
-    if response.code != '201'
-      @logger.debug("Failed to flush message via http with: #{response.code} - #{response.msg}") unless @logger.nil?
+    @pool.post do
+      begin
+        response = @http.send_request('PUT', '/tel/v2.0/metrics', message, headers)
+
+        if response.code != '201'
+          @logger.debug("Failed to flush message via http with: #{response.code} - #{response.msg}") unless @logger.nil?
+        end
+      rescue SocketError => ex
```

Comment:
Unfortunately `Net::HTTP` can throw a gazillion of exceptions, although not a great habit I'd suggest rescuing from `StandardError`. 

Considering our own code in the try block is pretty straightforward I think that'd be the best option to make sure the client doesn't throw any unhandled exception.